web/router.ex

```diff
@@ -43,20 +43,23 @@ defmodule Pairmotron.Router do
   end
 
   scope "/", Pairmotron do
-    pipe_through [:browser, :authenticate] # Use the default browser stack
+    pipe_through [:browser, :authenticate]
 
-    get "/pairs", PairController, :index
     get "/profile", ProfileController, :show
     get "/profile/edit", ProfileController, :edit
     put "/profile/update/:id", ProfileController, :update
     resources "/projects", ProjectController
+
+    get "/pairs", PairController, :index
+    get "/pairs/:year/:week", PairController, :show
+
+    get "/groups/:id/pairs", GroupPairController, :show
+    get "/groups/:id/pairs/:year/:week", GroupPairController, :show
+    delete "/groups/:id/pairs/:year/:week", GroupPairController, :delete
```

Comment:
The `:id` is what allowed canary to match on a `%Group{}`. I agree that `:group_id` is more specific, but I think `:id` matches the general pattern. These are really routes for groups first, then routes for pairs. If we could go the embedded resources way, it would still come through as `:id`.